server/py/services/alerts/crud/alerts.py

```diff
@@ -391,6 +404,27 @@ def reset_alert(self, session: sqlalchemy.orm.Session, project: str, name: str):
                 f"Alert {name} for project {project} does not exist"
             )
 
+        if alert.reset_policy == mlrun.common.schemas.alert.ResetPolicy.MANUAL:
+            # before resetting an alert, update number_of_events if required
+            number_of_events = self.get_number_of_events(alert.id)
+            if number_of_events > alert.criteria.count:
+                alert_state = (
+                    framework.utils.singletons.db.get_db().get_alert_state_dict(
+                        session, alert.id
+                    )
+                )
+                activation_time = alert_state.get("last_updated")
+                activation_id = alert_state.get("full_object", {}).get(
+                    "last_activation_id"
+                )
+                if activation_time and activation_id:
+                    framework.utils.singletons.db.get_db().update_alert_activation(
+                        session,
+                        activation_id=activation_id,
+                        activation_time=activation_time,
+                        number_of_events=number_of_events,
+                    )
```

Comment:
Maybe we can do it in a transaction with the `store_alert_state` call in the line afterwards?
Trying to minimize some db calls..

Reply:
@TomerShor it is done only in the cases where number_or_alerts is higher that count in criteria, so not really sure if it is worth it